docs/framework/wcf/samples/custom-lifetime.md

```diff
@@ -1,219 +1,227 @@
-  
- There is no way to renew the running timer when a new message arrives for the instance being moved to the idle state.  
-  
- The sample implements <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider> to intercept the calls to the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider.IsIdle%2A> method and route them to the `CustomLeaseExtension`. The <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider> implementation is contained in `CustomLifetimeLease` class. The <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider.IsIdle%2A> method is invoked when WCF is about to release the service instance. However, there is only one instance of a particular `ISharedSessionInstance` implementation in the ServiceBehavior’s <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider> collection. This means there is no way of knowing the <xref:System.ServiceModel.InstanceContext> being closed at the time WCF checks the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider.IsIdle%2A> method. Therefore this sample uses thread locking to serialize requests to the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider.IsIdle%2A> method.  
-  
+> The setup procedure and build instructions for this sample are located at the end of this article.
+
+## Shared instancing
+
+WCF offers several instancing modes for your service instances. The shared instancing mode covered in this article provides a way to share a service instance between multiple channels. Clients can contact a factory method in the service and create a new channel to start communication. The following code snippet shows how a client application creates a new channel to an existing service instance:
+
+```csharp
+// Create the first channel.
+IEchoService proxy = channelFactory.CreateChannel();
+
+//...
+
+// Close the first channel
+((IChannel)proxy).Close();
+
+// Send another request from a different channel using the same instance.
+proxy = channelFactory.CreateChannel();
+```
+
+Unlike other instancing modes, the shared instancing mode has a unique way of releasing the service instances. When all the channels are closed for an instance, the service WCF runtime starts a timer. If nobody makes a connection before the timeout expires, WCF releases the instance and claims the resources. As a part of the teardown procedure WCF invokes the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider.IsIdle%2A> method of all <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider> implementations before releasing the instance. If all of them return `true` the instance is released. Otherwise the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider> implementation is responsible for notifying the `Dispatcher` of the idle state by using a callback method.
+
+By default, the idle timeout value of <xref:System.ServiceModel.InstanceContext> is one minute. However, this sample demonstrates how you can extend this using a custom extension.
```

Comment:
I'm not sure about this comment, I'll double check and update once I know if this is true or not.

Reply:
This sentence implies that WCF will normally wait for one minute before closing an InstanceContext and that this sample shows how to extend that, but this isn't the case. WCF will immediately close an InstanceContext as soon as no channels are using it. This seems like a design change before WCF was released similar to how the sample changed between the document being written and WCF being released. There's a bunch of ambiguity with this sentence and the previous paragraph. I'll attempt to rewrite, feel free to correct grammar and formatting etc:

Unlike other instancing modes, the shared instancing mode has a unique way of releasing the service instances. By default, when all the channels are closed for an <xref:System.ServiceModel.InstanceContext>, the WCF service runtime checks to see if the service <xref:System.ServiceModel.InstanceContextMode> is configured to <xref:System.ServiceModel.InstanceContextMode.PerCall> or <xref:System.ServiceModel.InstanceContextMode.PerSession>, and if so releases the instance and claims the resources. If a custom <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider> is being used, WCF invokes the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider.IsIdle> method of the provider implementation before releasing the instance. If <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider.IsIdle> returns `true` the instance is released, otherwise the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider> implementation is responsible for notifying the `Dispatcher` of the idle state by using a callback method. This is done by calling the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider.NotifyIdle> method of the provider.

This sample demonstrates how you can delay releasing the <xref:System.ServiceModel.InstanceContext> with an idle timeout of 20 seconds.